pkg/operator/cluster/controller.go

```diff
@@ -119,6 +120,11 @@ func (c *ClusterController) onAdd(obj interface{}) {
 		c.devicesInUse = true
 	}
 
+	if cluster.Spec.MonSize <= 0 {
+		logger.Warningf("mon size is 0 or less (given: %d), needs to be greater 0. defaulting to %d", cluster.Spec.MonSize, clusterDefaultMonSize)
```

Comment:
I rather error here than go with default. The cluster manifest says 0 which is invalid. So it should fail.

Reply:
@kokhang The above condition (https://github.com/rook/rook/pull/1096/files/0c2329719f75407ea4d315cf9a25d3f5ae914993#diff-280625762ec1aa0553d19cf98117cf24R116) also doesn't fail.

Reply:
the error philosophy for CRDs that seems appropriate is:
- If there is an invalid value and we can pick a reasonable value, log a warning and pick the default
- If there is an invalid value and there is no reasonable value, log an error and abort creating the crd
- With K8s we can add CRD validation. For this setting, I would propose we want the values to be between 1-9, with a default of 3. If an even value is used, it should be a warning.

Reply:
I would always error on the side of logging an error and aborting when the input is invalid. The users rarely (if ever) look at logs so a warning log would not do much.

Reply:
@kokhang what do we gain by failing here? If we fail, this becomes a breaking change to the cluster.yaml. If we continue using the default of 3, existing cluster.yaml files will not break. There is no bad side effect of the default, so why wouldn't we pick the default? We are just asking for user headaches.